### PR TITLE
Remove the self-update command

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR:

- [x] Removes the self-update command from infection, since it no longer works
- [x] Doc PR: https://github.com/infection/site/pull/129

Fixes: #684, #425

### CODE REVIEW

**Comment by theofidry** [comment]
@BackEndTea any way to get this merge ASAP?

**Comment by BackEndTea** [comment]
@theofidry I'll get around to this somewhere this evening, if everything goes well

File: tests/AutoReview/ProjectCodeTest.php [hunk 1]
```diff
@@ -129,7 +128,6 @@ final class ProjectCodeTest extends TestCase
     private static $nonTestedConcreteClasses = [
         ConfigureCommand::class,
         InfectionCommand::class,
-        SelfUpdateCommand::class,
```
Comment (BackEndTea) [inline tests/AutoReview/ProjectCodeTest.php]: Note to self: The `padraic/phar-updater` dependency can also be removed now, fixing #425

**Comment by maks-rafalko** [comment]
Thank you @BackEndTea

**Comment by BackEndTea** [comment]
Thank you @maks-rafalko, @sanmai & @theofidry

**Pull request merged**